predicators/utils.py

```diff
@@ -922,7 +923,8 @@ def run_policy(
     last action from the returned trajectory to maintain the invariant that
     the trajectory states are of length one greater than the actions.
     """
-    state = env.reset(train_or_test, task_idx)
+    state = init_state if init_state is not None else env.reset(
```

Comment:
thinking out loud... if `env.reset` is not called, we're assuming that the environment's internal state here is equal to `init_state`, which should be true in the place that you're invoking it. however if this function was called by a different user who passed in an arbitrary `init_state`, then the environment would be out of sync with the `init_state` and there wouldn't be any error raised or anything, so it would get quite confusing.

ideally we would be able to do something like `assert env.get_state().allclose(init_state)` but we don't have a `get_state` method on environments. I suppose we could add one?

if we do add that method, then we might as well just call `state = env.get_state()` here instead of passing the `init_state` in as an argument

I can't immediately think of any better way, let me know if you have thoughts